src/ring_theory/jacobson.lean

```diff
@@ -252,4 +259,157 @@ end
+  {Rₘ Sₘ : Type*} [comm_ring Rₘ] [comm_ring Sₘ] [is_jacobson R]
+  (φ : R →+* S) (hφ : function.injective φ) (x : R) (hx : x ≠ 0)
+  (ϕ : localization_map (submonoid.powers x) Rₘ)
+  (ϕ' : localization_map ((submonoid.powers x).map φ : submonoid S) Sₘ)
+  (hφ' : (ϕ.map ((submonoid.powers x).mem_map_of_mem (φ : R →* S)) ϕ').is_integral) :
+  (⊥ : ideal S).jacobson = ⊥ :=
+begin
+  have hM : ((submonoid.powers x).map φ : submonoid S) ≤ non_zero_divisors S :=
+    map_le_non_zero_divisors_of_injective hφ
+      (le_non_zero_divisors_of_domain (λ h0, hx (let ⟨n, hn⟩ := h0 in pow_eq_zero hn))),
+  letI : integral_domain Sₘ := localization_map.integral_domain_of_le_non_zero_divisors ϕ' hM,
+  let φ' : Rₘ →+* Sₘ := ϕ.map ((submonoid.powers x).mem_map_of_mem (φ : R →* S)) ϕ',
+  suffices : ∀ I : ideal Sₘ, I.is_maximal → (I.comap ϕ'.to_map).is_maximal,
+  { have hϕ' : comap ϕ'.to_map ⊥ = ⊥ :=
+    by simpa [ring_hom.injective_iff_ker_eq_bot, ring_hom.ker_eq_comap_bot] using (ϕ'.injective hM),
+    refine eq_bot_iff.2 (le_trans _ (le_of_eq hϕ')),
+    have hSₘ : is_jacobson Sₘ := is_jacobson_of_is_integral' φ' hφ' (is_jacobson_localization ϕ),
+    rw [← hSₘ ⊥ radical_bot_of_integral_domain, comap_jacobson],
+    exact Inf_le_Inf (λ j hj, ⟨bot_le, let ⟨J, hJ⟩ := hj in hJ.2 ▸ this J hJ.1.2⟩) },
+  introsI I hI,
+  -- Remainder of the proof is pulling and pushing ideals around the square and the quotient square
+  haveI : (I.comap ϕ'.to_map).is_prime := comap_is_prime ϕ'.to_map I,
+  haveI : (I.comap φ').is_prime := comap_is_prime φ' I,
+  haveI : (⊥ : ideal (I.comap ϕ'.to_map).quotient).is_prime := bot_prime,
+  have hcomm: φ'.comp ϕ.to_map = ϕ'.to_map.comp φ := ϕ.map_comp _,
+  let f := quotient_map (I.comap ϕ'.to_map) φ le_rfl,
+  let f' := quotient_map I φ' le_rfl,
+  let g := quotient_map I ϕ'.to_map le_rfl,
+  let g' : ((I.comap ϕ'.to_map).comap φ).quotient →+* (I.comap φ').quotient :=
+    quotient_map (I.comap φ') ϕ.to_map
+    (le_of_eq (trans (comap_comap φ ϕ'.to_map) (hcomm ▸ (comap_comap ϕ.to_map φ').symm))),
+  have : ((I.comap ϕ'.to_map).comap φ).is_maximal := (by rw [comap_comap, hcomm, ← comap_comap] :
+    ((I.comap φ').comap ϕ.to_map) = ((I.comap ϕ'.to_map).comap φ)) ▸
+      ((is_maximal_iff_is_maximal_disjoint ϕ _).1
+      (is_maximal_comap_of_is_integral_of_is_maximal' φ' hφ' I hI)).left,
+  rw is_maximal_iff_bot_quotient_is_maximal at this ⊢,
+  refine is_maximal_of_is_integral_of_is_maximal_comap' f _ ⊥
+    ((eq_bot_iff.2 (comap_bot_le_of_injective f quotient_map_injective)).symm ▸ this),
+  exact is_integral_tower_bot_of_is_integral' f g quotient_map_injective
+    ((comp_quotient_map_eq_of_comp_eq hcomm I).symm ▸
+    (ring_hom.is_integral_trans (is_integral_of_surjective'
+      (localization_map.surjective_quotient_map_of_maximal_of_localization
+      (by rwa [comap_comap, hcomm, is_maximal_iff_bot_quotient_is_maximal])))
+      (is_integral_quotient_of_is_integral' hφ'))),
+end
```

Comment:
A little bit of golfing, although I don't know enough about Jacobson rings to say whether I missed an extra shortcut.
```suggestion
begin
  have hM : ((submonoid.powers x).map φ : submonoid S) ≤ non_zero_divisors S :=
    map_le_non_zero_divisors_of_injective hφ (powers_le_non_zero_divisors_of_domain hx),
  letI : integral_domain Sₘ := localization_map.integral_domain_of_le_non_zero_divisors ϕ' hM,
  let φ' : Rₘ →+* Sₘ := ϕ.map ((submonoid.powers x).mem_map_of_mem (φ : R →* S)) ϕ',
  suffices : ∀ I : ideal Sₘ, I.is_maximal → (I.comap ϕ'.to_map).is_maximal,
  { have hϕ' : comap ϕ'.to_map ⊥ = ⊥,
    { simpa [ring_hom.injective_iff_ker_eq_bot, ring_hom.ker_eq_comap_bot] using ϕ'.injective hM },
    refine eq_bot_iff.2 (le_trans _ (le_of_eq hϕ')),
    have hSₘ : is_jacobson Sₘ := is_jacobson_of_is_integral' φ' hφ' (is_jacobson_localization ϕ),
    rw [← hSₘ ⊥ radical_bot_of_integral_domain, comap_jacobson],
    exact Inf_le_Inf (λ j hj, ⟨bot_le, let ⟨J, hJ⟩ := hj in hJ.2 ▸ this J hJ.1.2⟩) },
  introsI I hI,
  -- Remainder of the proof is pulling and pushing ideals around the square and the quotient square
  haveI : (I.comap ϕ'.to_map).is_prime := comap_is_prime ϕ'.to_map I,
  haveI : (I.comap φ').is_prime := comap_is_prime φ' I,
  haveI : (⊥ : ideal (I.comap ϕ'.to_map).quotient).is_prime := bot_prime,
  have hcomm: φ'.comp ϕ.to_map = ϕ'.to_map.comp φ := ϕ.map_comp _,
  let f := quotient_map (I.comap ϕ'.to_map) φ le_rfl,
  let f' := quotient_map I φ' le_rfl,
  let g := quotient_map I ϕ'.to_map le_rfl,
  let g' : ((I.comap ϕ'.to_map).comap φ).quotient →+* (I.comap φ').quotient :=
    quotient_map (I.comap φ') ϕ.to_map
    (le_of_eq (trans (comap_comap φ ϕ'.to_map) (hcomm ▸ (comap_comap ϕ.to_map φ').symm))),
  have := ((is_maximal_iff_is_maximal_disjoint ϕ _).1
    (is_maximal_comap_of_is_integral_of_is_maximal' φ' hφ' I hI)).left,
  have : ((I.comap ϕ'.to_map).comap φ).is_maximal,
  { rwa [comap_comap, hcomm, ← comap_comap] at this },
  rw is_maximal_iff_bot_quotient_is_maximal at this ⊢,
  refine is_maximal_of_is_integral_of_is_maximal_comap' f _ ⊥
    ((eq_bot_iff.2 (comap_bot_le_of_injective f quotient_map_injective)).symm ▸ this),
  exact is_integral_tower_bot_of_is_integral' f g quotient_map_injective
    ((comp_quotient_map_eq_of_comp_eq hcomm I).symm ▸
    (ring_hom.is_integral_trans (is_integral_of_surjective'
      (localization_map.surjective_quotient_map_of_maximal_of_localization
      (by rwa [comap_comap, hcomm, is_maximal_iff_bot_quotient_is_maximal])))
      (is_integral_quotient_of_is_integral' hφ'))),
end
```